kitty-items-js/src/services/market.ts

```diff
@@ -121,24 +158,42 @@ class MarketService {
     });
   };
 
-  findMostRecentSales = async () => {
-    return SaleOffer.query().orderBy("created_at", "desc").limit(20);
+  addSaleOffer = async (saleOfferEvent) => {
+    return SaleOffer.transaction(async (tx) => {
+      return await SaleOffer.query(tx).insert({
+        saleItemId: saleOfferEvent.data.saleItemID,
+        saleItemType: saleOfferEvent.data.saleItemType,
+        saleItemCollection: saleOfferEvent.data.saleItemCollection,
+        transactionId: saleOfferEvent.transactionId,
+        price: saleOfferEvent.data.price,
+      });
+    });
   };
 
-  upsertSaleOffer = async (itemId: number, price: number) => {
+  removeSaleOffer = (saleOfferEvent) => {
     return SaleOffer.transaction(async (tx) => {
-      const saleOffers = await SaleOffer.query(tx).insertGraphAndFetch([
-        {
-          kitty_item: {
-            id: itemId,
-          },
-          price: price,
-          is_complete: false,
-        },
-      ]);
-      return saleOffers[0];
+      return await SaleOffer.query(tx).where({
+        saleItemId: saleOfferEvent.data.saleItemID,
+      }).del()
     });
   };
+
+  findMostRecentSales = () => {
+    // TODO
```

Comment:
Does this not working anymore?

```
return SaleOffer.query().orderBy("created_at", "desc").limit(20);
```

Reply:
Should work. This method was implemented for the old SaleOffer table so I may have erased it (Thought I was going to be able to get by just storing events). Can re-add it.